src/app/core/rest/document.service.ts

```diff
@@ -64,6 +64,7 @@ export class DocumentService {
     return this.httpClient.put<Document>(`${this.apiPrefix(document.collectionCode)}/${document.id}/data`, document.data)
       .pipe(
         catchError(error => this.handleGlobalError(error)),
+        tap(returnedDocument => returnedDocument.collectionCode = document.collectionCode),
```

Comment:
I don't understand this change. You basically change the document outside of the pipeline. The correct implementation of your intentions would be using the `map` operator:
```typescript
map(returnedDocument => {...returnedDocument, collectionCode: document.collectionCode}),
```
However, it's still not what we want here. Why isn't `collectionCode` returned from the server?